### PR TITLE
ensure that sessions on the action plan progress page are always pulled from the latest approved action plan

### PR DESCRIPTION
**Pull request by tomsmyers**
https://trello.com/c/xN23pxkV/175-bug-fix-sessions-disppears-when-action-plan-is-in-draft-state

## What does this pull request do?

ensure that sessions on the action plan progress page are always pulled from the latest approved action plan

## What is the intent behind these changes?

stop sessions going missing when the SP creates a new draft action plan

### CODE REVIEW

File: server/routes/serviceProviderReferrals/serviceProviderReferralsController.ts
Comment (sldblog) [inline server/routes/serviceProviderReferrals/serviceProviderReferralsController.ts:201]: 💭 The usage is always `approvedActionPlans.sort(sortBy('-approvedAt'))?.[0]` -- since the older ones are always unused, do we need the list endpoint? Could we have a singular endpoint instead? (YAGNI)

**Comment by sldblog** [comment]
> stop sessions going missing

How are sessions missing though? Don't we migrate the appointments to the new AP?

File: server/routes/serviceProviderReferrals/serviceProviderReferralsController.ts
Comment (tomsmyers) [reply server/routes/serviceProviderReferrals/serviceProviderReferralsController.ts:201]: this endpoint is used elsewhere, and i think adding a new endpoint just for this use case seems unnecessary.

**Comment by tomsmyers** [comment]
> rate the appointments to the new AP?

have a look in the ticket for more info. but the sessions dissapear from the UI because the draft AP does not get sessions associated with it until approval (where any additional sessions are also added). happy to give a demo.

also, this change fundamentally makes sense. be it new unapproved APs, or new draft APs, or any unexpected situation, the appointments listed on this page should always come from the latest approved AP.

**Review by sldblog — APPROVED** [review]
Thanks -- makes sense 👍

File: server/routes/serviceProviderReferrals/serviceProviderReferralsController.ts
Comment (sldblog) [reply server/routes/serviceProviderReferrals/serviceProviderReferralsController.ts:201]: ah did not know -- thanks. Checked its usage and it's used as a list. All good 👍

**Pull request merged**
